src/components/UI/TodoTable.tsx

```diff
@@ -6,20 +6,29 @@ import {
   TableCell,
   TableBody,
   Paper,
+  Alert,
+  LinearProgress,
 } from "@mui/material";
+import { useMutation, useQuery } from "react-query";
+
+import { deleteTodo } from "../../services/queries/delete-todo";
+import Todo from "../../models/todo";
+
+const TodoTable: ({ todos }: { todos: Todo[] }) => JSX.Element = ({
+  todos,
+}) => {
+  const { mutate } = useMutation(deleteTodo);
+
+  const fetchTodo = () => {
+    return todos;
+  };
+
+  const { data, isLoading, isError } = useQuery("todos", fetchTodo);
+
+  if (isLoading) return <LinearProgress color="secondary" />;
+
+  if (isError) return <Alert severity="error">Something went wrong</Alert>;
```

Comment:
I assume these states are never triggered as we are not loading them really, just returning the props todos. Also don't do inline return

Reply:
loading is hit for a sec, but it's not necessary